node/src/indexer/fake.rs

```diff
@@ -678,8 +731,9 @@ impl FakeIndexerManager {
 
     /// Adds a new node to the fake indexer. Returns the API for the node, a task that
     /// runs the node's logic, and the running job name to passed to the coordinator.
-    pub fn add_indexer_node(
+    pub async fn add_indexer_node(
```

Comment:
From what I can see the only reason to make this async now is due to the use of a `tokio::sync::Mutex` over the `std` one. I'd recommend to go with the `std` mutex here instead.

Reply:
Also this is a good read for more context: https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use